modin/pandas/base.py

```diff
@@ -3111,21 +3111,53 @@ def __init__(
         axis = self._dataframe._get_axis_number(axis)
         # FIXME: this should be converted into a dict to ensure simplicity
         # of handling resample parameters at the query compiler level.
-        self.resample_args = [
-            rule,
-            axis,
-            closed,
-            label,
-            convention,
-            kind,
-            loffset,
-            base,
-            on,
-            level,
-            origin,
-            offset,
-        ]
-        self.__groups = self.__get_groups(*self.resample_args)
+        self.resample_args = {
+            "rule": rule,
+            "axis": axis,
+            "closed": closed,
+            "label": label,
+            "convention": convention,
+            "kind": kind,
+            "loffset": loffset,
+            "base": base,
+            "on": on,
+            "level": level,
+            "origin": origin,
+            "offset": offset,
+        }
+        self.__groups = self.__get_groups(*list(self.resample_args.values()))
+
+    def __getitem__(self, key):
+        if len(self._dataframe.columns.intersection(key)) != len(key):
+            missed_keys = list(set(key).difference(self._dataframe.columns))
+            raise KeyError(f"Columns {str(missed_keys)[1:-1]} don't exist.")
+        return self._gotitem(list(key), ndim=2)
+
+    def _gotitem(self, key, ndim: int):
+        """
+        Get sub-classes to define.
+
+        Parameters
+        ----------
+        key : str or list
+            String or list of selections.
+        ndim : {1, 2}
+            Requested ndim of result.
+
+        Returns
+        -------
+        modin.pandas.BasePandasDataset
+            Sliced object.
+        """
+        kwargs = self.resample_args
+
+        try:
+            subset_ = self._dataframe[key]
+        except IndexError:
+            subset_ = self._dataframe
```

Comment:
Why does we set the full `self._dataframe` to subset if dataframe doesn't contain `key` column? Does pandas make the same?

Reply:
@Rubtsowa give explanation related this please to resolve this conversation

Reply:
This case using in pandas